rspirv/dr/constructs.rs

```diff
@@ -135,6 +132,70 @@ impl Module {
             .chain(&self.annotations)
             .chain(&self.types_global_values)
     }
+
+    /// Returns a mut iterator over all global instructions.
+    pub fn global_inst_iter_mut(&mut self) -> impl Iterator<Item = &mut Instruction> {
+        self.capabilities
+            .iter_mut()
+            .chain(&mut self.extensions)
+            .chain(&mut self.ext_inst_imports)
+            .chain(&mut self.memory_model)
+            .chain(&mut self.entry_points)
+            .chain(&mut self.execution_modes)
+            .chain(&mut self.debugs)
+            .chain(&mut self.annotations)
+            .chain(&mut self.types_global_values)
+    }
+
+    /// Returns a iterator over all instructions.
+    pub fn all_inst_iter(&self) -> impl Iterator<Item = &Instruction> {
+        self.capabilities
+            .iter()
+            .chain(&self.extensions)
+            .chain(&self.ext_inst_imports)
+            .chain(&self.memory_model)
+            .chain(&self.entry_points)
+            .chain(&self.execution_modes)
+            .chain(&self.debugs)
+            .chain(&self.annotations)
+            .chain(&self.types_global_values)
+            .chain(self.functions.iter().flat_map(|f| {
+                f.def
+                    .iter()
+                    .chain(f.parameters.iter())
+                    .chain(
+                        f.blocks
+                            .iter()
+                            .flat_map(|b| b.label.iter().chain(b.instructions.iter())),
```

Comment:
took me a bit to remember `Option` is `IntoIterator` for the label here :P

Reply:
Yeah it's nice Option can do that, but it's not always intuitive.

Reply:
Monads :tada: